src/features/applicationView/offerCard/sendInvoiceForm/__tests__/SendInvoiceForm.test.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { act } from 'react-dom/test-utils';
+
+import SendInvoiceForm, { SendInvoiceFormProps } from '../SendInvoiceForm';
+
+const mockProps: SendInvoiceFormProps = {
+  email: 'test@example.com',
+  isSubmitting: false,
+  onCancel: jest.fn(),
+  onSubmit: jest.fn(),
+};
+
+const realDateNow = Date.now.bind(global.Date);
+
+beforeAll(() => {
+  global.Date.now = jest.fn(() => new Date('2020-09-23T00:00:00.000Z').valueOf());
+});
+
+afterAll(() => {
+  global.Date.now = realDateNow;
+});
+
+describe('SendInvoiceForm', () => {
+  const getWrapper = (props?: Partial<SendInvoiceFormProps>) => mount(<SendInvoiceForm {...mockProps} {...props} />);
+
+  it('renders normally', () => {
+    const wrapper = getWrapper();
+    expect(wrapper.render()).toMatchSnapshot();
+  });
+
+  it('shows an error if email is missing', () => {
+    const wrapper = getWrapper({ email: null });
+    expect(wrapper.find('Text.missingEmail')).toMatchInlineSnapshot(`
+      <Text
+        className="missingEmail"
+        color="critical"
+      >
+        <span
+          className="text critical span missingEmail"
+        >
+          Sähköpostiosoite puuttuu
+        </span>
+      </Text>
+    `);
+  });
+
+  it('calls onSubmit on submit', async () => {
+    const onSubmit = jest.fn();
+    const wrapper = getWrapper({ onSubmit });
+    await act(async () => {
+      wrapper.find('Form').simulate('submit');
```

Comment:
With testing library:

userEvent.click(screen.getByRole('button', { name: 'Lähetä' }).

with that you are actually testing that there is submit button with correct text and it actually triggers submit when clicked. Just like user would actually do it :)

Reply:
The idea is for the test to very closely represent how the user uses the form

Reply:
Good blog post on testing implementation details in react:

https://kentcdodds.com/blog/testing-implementation-details